thentos-core/src/Thentos/Types/Opaque.hs

```diff
@@ -0,0 +1,24 @@
+-- | Types whose internal representation is not accessible.
```

Comment:
Is this module just a workaround for the missing export list in `Types.hs`. If so, I'd prefer to have an export list in `Types.hs` and move these types there.